src/webui/api/torrentscontroller.cpp

```diff
@@ -277,6 +278,50 @@ namespace
         return trackerList;
     }
 
+    QJsonArray getFiles(const BitTorrent::Torrent *const torrent, QList<int> fileIndexes = {})
+    {
+        Q_ASSERT(torrent->hasMetadata());
+        if (!torrent->hasMetadata()) [[unlikely]]
+            return {};
+        
+        if (fileIndexes.isEmpty())
+        {
+            const int filesCount = torrent->filesCount();
+            fileIndexes.reserve(filesCount);
+            for (int i = 0; i < filesCount; ++i)
+                fileIndexes.append(i);
+        }
+
+        QJsonArray fileList;
+        const QList<BitTorrent::DownloadPriority> priorities = torrent->filePriorities();
+        const QList<qreal> fp = torrent->filesProgress();
+        const QList<qreal> fileAvailability = torrent->fetchAvailableFileFractions().takeResult();
+        const BitTorrent::TorrentInfo info = torrent->info();
+        for (const int index : asConst(fileIndexes))
+        {
+            QJsonObject fileDict =
+            {
+                {KEY_FILE_INDEX, index},
+                {KEY_FILE_PROGRESS, fp[index]},
+                {KEY_FILE_PRIORITY, static_cast<int>(priorities[index])},
+                {KEY_FILE_SIZE, torrent->fileSize(index)},
+                {KEY_FILE_AVAILABILITY, fileAvailability[index]},
+                // need to provide paths using a platform-independent separator format
+                {KEY_FILE_NAME, torrent->filePath(index).data()}
+            };
+
+            const BitTorrent::TorrentInfo::PieceRange idx = info.filePieces(index);
+            fileDict[KEY_FILE_PIECE_RANGE] = QJsonArray {idx.first(), idx.last()};
```

Comment:
This can be streamlined if you are interested:
```suggestion
            const BitTorrent::TorrentInfo::PieceRange idx = info.filePieces(index);
            
            QJsonObject fileDict =
            {
                {KEY_FILE_INDEX, index},
                {KEY_FILE_PROGRESS, fp[index]},
                {KEY_FILE_PRIORITY, static_cast<int>(priorities[index])},
                {KEY_FILE_SIZE, torrent->fileSize(index)},
                {KEY_FILE_AVAILABILITY, fileAvailability[index]},
                // need to provide paths using a platform-independent separator format
                {KEY_FILE_NAME, torrent->filePath(index).data()},
                {KEY_FILE_PIECE_RANGE, QJsonArray {idx.first(), idx.last()}
            };
```